lib/util/device-service-details.ts

```diff
@@ -0,0 +1,135 @@
+import groupBy = require('lodash/groupBy');
+import mapValues = require('lodash/mapValues');
+import omit = require('lodash/omit');
+import {
+	CurrentGatewayDownload,
+	CurrentService,
+	DeviceWithImageInstalls,
+	DeviceWithServiceDetails,
+	GatewayDownload,
+	Image,
+	ImageInstall,
+	PineOptionsFor,
+	Release,
+	Service,
+} from '../../typings/balena-sdk';
+
+// Pine options necessary for getting raw service data for a device
+export const getCurrentServiceDetailsPineOptions = (expandRelease: boolean) => {
+	const pineOptions: PineOptionsFor<DeviceWithImageInstalls> = {
+		$expand: {
+			image_install: {
+				$select: ['id', 'download_progress', 'status', 'install_date'],
+				$filter: {
+					status: {
+						$ne: 'deleted',
+					},
+				},
+				$expand: {
+					image: {
+						$select: ['id'],
+						$expand: {
+							is_a_build_of__service: {
+								$select: ['id', 'service_name'],
+							},
+						},
+					},
+					...(expandRelease && {
+						is_provided_by__release: {
+							$select: ['id', 'commit'],
+						},
+					}),
+				},
+			},
+			gateway_download: {
+				$select: ['id', 'download_progress', 'status'],
+				$filter: {
+					status: {
+						$ne: 'deleted',
+					},
+				},
+				$expand: {
+					image: {
+						$select: ['id'],
+						$expand: {
+							is_a_build_of__service: {
+								$select: ['id', 'service_name'],
+							},
+						},
+					},
+				},
+			},
+		},
+	};
+
+	return pineOptions;
+};
+
+function getSingleInstallSummary(rawData: ImageInstall): CurrentService;
+function getSingleInstallSummary(
+	rawData: GatewayDownload,
+): CurrentGatewayDownload;
+function getSingleInstallSummary(
+	rawData: ImageInstall | GatewayDownload,
+): CurrentService | CurrentGatewayDownload {
+	const image = (rawData.image as Image[])[0];
+	const service = (image.is_a_build_of__service as Service[])[0];
+
+	let releaseInfo: { commit?: string } = {};
+	if (
+		'is_provided_by__release' in rawData &&
+		rawData.is_provided_by__release != null
+	) {
+		const release = (rawData.is_provided_by__release as Release[])[0];
+		releaseInfo = {
+			commit: release != null ? release.commit : void 0,
```

Comment:
I believe using `undefined` is fine as well for all the browsers we support and non-deprecated NodeJS versions, but I am fine with it as it is as well.

Reply:
Ooops. I missed that part of decaffeinate :sweat_smile: 
I will fix it in the follow-up PR, so save some time.
Thanks for spotting it :+1: